cumulus/tasks/discover-pdrs/tests/index.js

```diff
@@ -222,3 +222,67 @@ test('test pdr discovery with SFTP assuming some PDRs are new', (t) => {
       return aws.recursivelyDeleteS3Bucket(internalBucketName).then(t.fail);
     });
 });
+
+test('test pdr discovery with S3 assuming some PDRs are new', async (t) => {
+  const provider = {
+    id: 'MODAPS',
+    protocol: 's3'
+  };
+
+  const newPayload = Object.assign({}, input);
+  newPayload.config.provider = provider;
+  newPayload.config.useQueue = false;
+  const pdrBucketName = testUtils.randomString();
+  newPayload.config.collection.provider_path = `s3://${pdrBucketName}/test-data/pdrs`;
+  newPayload.input = {};
+
+  // upload test data to s3
+  const pdrs = [
+    'MOD09GQ.PDR',
+    'MYD13A1_5_grans.PDR',
+    'this_is_not_PDR_file',
+    'PDN.ID1611071307.PDR',
+    'PDN.ID1611081200.PDR'
+  ];
+
+  await aws.s3().createBucket({ Bucket: pdrBucketName }).promise()
+    .then(() => {
```

Comment:
Since this is an async function, there should not be instances of `.then()`.  Related to what we were talking about this morning.

Reply:
done